stop calling findOptions recursively, and be stubborn when trying to require the linter.

gustavnikolaj opened the pull request:
extends #22 

This is a bit messy, but it follows up from changes caused by #18 and tries to improve the problems described in #19 

First, we stop at the first `package.json` file we find. Meaning that we won't use a parent modules linter configuration when linting a file in a nested module.

Try to require the linter in the lintWorker until it succeeds. Without this, the lintWorker would crash and cause annoying errors when trying to lint with a linter that hadn't been installed yet.

gustav-olsen-groupone committed:
simplify tests

gustav-olsen-groupone committed:
don't call findOptions recursively when no linter is found

gustav-olsen-groupone committed:
try to load linter until it succeeds, and suppress errors

novemberborn commented:
Heads up, I changed the base so the comparison is against #22. It makes it easier to review the changes just in this PR.

gustavnikolaj commented:
I rebased it on master and force pushed the branch. I changed the base after pushing the new commits, which I think upset the CI systems. Trying again :-)
